Create separate SPM packages

guykogus opened the pull request:
🚀

- Creates a separate SPM package for each library extension.
- Maintains the original `import SwifterSwift` usage so that there weren't be issues with backwards compatibility.
- Removes the old Xcode project and workspace and change CI tests to use the Swift packages, since they can now run tests on all platforms (including watchOS!)

Fixes #1023

## Checklist
<!--- Please go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x] New extensions are written in Swift 5.6.
- [x] New extensions support iOS 12.0+ / tvOS 12.0+ / macOS 10.13+ / watchOS 4.0+, or use `@available` if not.
- [x] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.
- [x] I have added a [changelog](https://github.com/SwifterSwift/SwifterSwift/blob/master/CHANGELOG_GUIDELINES.md) entry describing my changes.

guykogus commented:
@LucianoPAlmeida what do you think? I was wondering if maybe we need to completely refactor it out so that each extension package exists in a separate folder. I.e. at the root of the project have `SwifterSwift-AppKit/Sources/*`. This seems to work, too, but it's another option.

LucianoPAlmeida commented:
> @LucianoPAlmeida what do you think? I was wondering if maybe we need to completely refactor it out so that each extension package exists in a separate folder. I.e. at the root of the project have `SwifterSwift-AppKit/Sources/*`. This seems to work, too, but it's another option.

I think this is fine if that works and such a refactor will be much work for no much value if the way it is serves the purpose.

guykogus committed:
Refactor project for SPM with separated targets

Made-with: Cursor

guykogus committed:
fix(spm): wire shared imports and stabilize SwiftPM tests

Add SwifterSwiftShared imports where shared typealiases are used, improve JSON resource lookup under SPM bundles, and make a few tests resilient to locale/timezone differences.

Made-with: Cursor

guykogus committed:
Revert logic

guykogus committed:
Fix tests for SPM

guykogus committed:
ci: add SwiftPM package tests matrix

Run SwifterSwift-Package scheme via xcodebuild for iOS/tvOS/macOS (and build for watchOS) alongside CocoaPods lint.

Made-with: Cursor

guykogus committed:
Fix project to use Folders instead of Groups

guykogus committed:
Fix compilation

guykogus committed:
Fix tests for watchOS

guykogus committed:
Add watchOS tests

guykogus committed:
Fix SPM tests

guykogus committed:
Fix tests

guykogus committed:
Fix tests

guykogus committed:
Removed old Xcode project and workspace

guykogus committed:
Remove old tests

guykogus committed:
Skip broken WKWebView test for iOS

guykogus committed:
Fix isNumeric

guykogus committed:
Revert deleted test

guykogus committed:
Fix WKWebView tests

guykogus committed:
Merge branch 'master' into spm-packages
